docs/getting-started.md

```diff
@@ -128,155 +131,163 @@ c.JupyterHub.hub_port = 54321
 c.JupyterHub.ssl_cert = '/path/to/my.cert'
 ```
 
+Some cert files also contain the key, in which case only the cert is needed. It is important that these files be put in a secure location on your server.
+
 There are two other aspects of JupyterHub network security.
-The Hub authenticates its requests to the proxy via an environment variable,
-`CONFIGPROXY_AUTH_TOKEN`. If you want to be able to start or restart the proxy
-or Hub independently of each other (not always necessary),
-you must set this environment variable before starting the server:
 
-```bash
-export CONFIGPROXY_AUTH_TOKEN=`openssl rand -hex 32`
+The cookie secret is encryption key, used to encrypt the cookies used for authentication. If this value
+changes for the Hub, all single-user servers must also be restarted. Normally, this value is stored in the
+file `jupyterhub_cookie_secret`, which can be specified with:
+
+```python
+c.JupyterHub.cookie_secret_file = '/path/to/jupyterhub_cookie_secret'
 ```
 
-If you don't set this, the Hub will generate a random key itself,
-which means that any time you restart the Hub you **must also restart the proxy**.
-If the proxy is a subprocess of the Hub, this should happen automatically.
+In most deployments of JupyterHub, you should point this to a secure location on the file system. If the
+cookie secret file doesn't exist when the Hub starts, a new cookie secret is generated and stored in the
+file.
 
-The cookie secret is another key, used to encrypt the cookies used for authentication.
-If this value changes for the Hub, all single-user servers must also be restarted.
-Normally, this value is stored in the file `jupyterhub_cookie_secret`, which can be specified with:
+If you would like to avoid the need for files, the value can be loaded from the `JPY_COOKIE_SECRET` env
+variable:
 
-```python
-c.JupyterHub.cookie_secret_file = '/path/to/cookie_secret'
+```bash
+export JPY_COOKIE_SECRET=`openssl rand -hex 1024`
 ```
 
-If the cookie secret file doesn't exist when the Hub starts,
-a new cookie secret is generated and stored in the file.
+The Hub authenticates its requests to the Proxy via an environment variable, `CONFIGPROXY_AUTH_TOKEN`. If
+you want to be able to start or restart the proxy or Hub independently of each other (not always
+necessary), you must set this environment variable before starting the server:
 
-If you would like to avoid the need for files,
-the value can be loaded from the `JPY_COOKIE_SECRET` env variable:
 
 ```bash
-export JPY_COOKIE_SECRET=`openssl rand -hex 1024`
+export CONFIGPROXY_AUTH_TOKEN=`openssl rand -hex 32`
 ```
 
+If you don't set this, the Hub will generate a random key itself, which means that any time you restart
+the Hub you **must also restart the Proxy**. If the proxy is a subprocess of the Hub, this should happen
+automatically (this is the default configuration).
+
+**(Min: does the cookie secret and auth token env vars need to be only visible by root?)** What if other users can see them?
```

Comment:
Auth token must be set for the Hub and the Proxy, cookie secret only for the Hub.